### PR TITLE
ZK-4597: Disabled Combobutton should not be able to focus or controlled by keyboard

### PR DESCRIPTION
**Pull request by b952426**
ZK-4597: Disabled Combobutton should not be able to focus or controlled by keyboard

### CODE REVIEW

File: zktest/src/archive/test2/B96-ZK-4597.zul [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+Copyright (C) 2021 Potix Corporation. All Rights Reserved.
+-->
+<zk>
+	<label>press tab(focus on combobutton) and press ArrowDown/Enter/Spacebar</label>
+	<combobutton id="disabled" label="popup" tabindex="0" iconSclass="z-icon-home" disabled="true" onClick='Clients.log("should not trigger if isDisabled")'>
+		<popup>
+			should not popup if isDisabled
+		</popup>
+	</combobutton>
+</zk>
```
Comment (scribetw) [inline zktest/src/archive/test2/B96-ZK-4597.zul:21]: Add a case to toggle disable dynamically

File: zktest/test/java/org/zkoss/zktest/zats/test2/B96_ZK_4597Test.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.openqa.selenium.Keys;
+import org.zkoss.zktest.zats.WebDriverTestCase;
+
+public class B96_ZK_4597Test extends WebDriverTestCase {
+	@Test
+	public void test() {
+		connect();
+		waitResponse();
+		getActions().sendKeys(Keys.TAB).sendKeys(Keys.SPACE).sendKeys(Keys.ENTER).sendKeys(Keys.ARROW_DOWN).perform();
+		Assert.assertFalse(jq("@textarea").exists());
```
Comment (scribetw) [inline zktest/test/java/org/zkoss/zktest/zats/test2/B96_ZK_4597Test.java:25]: isZKLogAvailable

File: zul/src/archive/web/js/zul/wgt/Combobutton.js [hunk 3]
```diff
@@ -233,6 +233,9 @@ zul.wgt.Combobutton = zk.$extends(zul.wgt.Button, {
 			this.$supers('doKeyDown_', arguments);
 	},
 	_doKeyDown: function (evt) { //support enter,space, arrow down, and escape
+		if (this.isDisabled) {
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/wgt/Combobutton.js:236]: isDisabled()

**Review by scribetw — CHANGES_REQUESTED** [review]
If disabled, don't add tabindex (init and setproperty)

File: zul/src/archive/web/js/zul/wgt/Combobutton.js [hunk 4]
```diff
@@ -89,7 +89,22 @@ zul.wgt.Combobutton = zk.$extends(zul.wgt.Button, {
...
+		 */
+		/** Sets whether it is disabled.
+		 * If disabled is true, user cannot tab into the element.
+		 * @param boolean disabled
+		 */
+		disabled: function (v) {
+			var n = this.$n();
+			if (n) {
+				jq(n).attr('disabled', v ? 'disabled' : null),
+				jq(n).attr('tabindex', v ? -1 : this._tabindex | 0);
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/wgt/Combobutton.js:105]: `v ? null : ...`

File: zul/src/archive/web/js/zul/wgt/Combobutton.js [hunk 5]
```diff
@@ -89,7 +89,22 @@ zul.wgt.Combobutton = zk.$extends(zul.wgt.Button, {
...
+		 * @return boolean
+		 */
+		/** Sets whether it is disabled.
+		 * If disabled is true, user cannot tab into the element.
+		 * @param boolean disabled
+		 */
+		disabled: function (v) {
+			var n = this.$n();
+			if (n) {
+				jq(n).attr('disabled', v ? 'disabled' : null),
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/wgt/Combobutton.js:104]: ;

File: zul/src/archive/web/js/zul/wgt/mold/combobutton.js [hunk 6]
```diff
@@ -18,10 +18,7 @@ function (out) {
 	
 	out.push('<span ', this.domAttrs_({tabindex: 1}));//we have a default 0 for tabindex of this element
 	
-	if (this._disabled)
-		out.push(' disabled="disabled" aria-disabled="true"');
-	
-	out.push(' tabindex="', tabi, '"');
+	this._disabled ? out.push(' disabled="disabled" aria-disabled="true"') : out.push(' tabindex="', tabi, '"');
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/wgt/mold/combobutton.js:21]: if ... else ...

**Review by scribetw — CHANGES_REQUESTED** [review]

**Review by scribetw — APPROVED** [review]

**Review by plijyfes — APPROVED** [review]

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**